storage/resource_mount.go

```diff
@@ -37,6 +39,34 @@ func (cb mountCallback) preProcess(r *schema.Resource) func(
 	}
 }
 
+// If a cluster_id is specified for the mount, this function validates that the
+// cluster actually exists. If the cluster does not exist, we assume (incorrectly)
+// that the mount also does not exist.
+//
+// In cases of update requests this would lead to an attempted recreation of the
+// mount
+// In cases of delete, this could orphan the mount
+//
+// This is needed to resolve issue: https://github.com/databricks/terraform-provider-databricks/issues/1864
+// because without a running cluster we do not have a way to do CRUD on mounts.
+// The argument being that possibly orphaning mounts is a better experience than
+// requiring manual edits to tfstate.
+func validateCluster(cb func(context.Context, *schema.ResourceData, any) diag.Diagnostics) func(context.Context, *schema.ResourceData, any) diag.Diagnostics {
+	return func(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+		clusterId := d.Get("cluster_id").(string)
+		if clusterId != "" {
+			clustersAPI := clusters.NewClustersAPI(ctx, m)
+			_, err := clustersAPI.Get(clusterId)
+			if apierr.IsMissing(err) {
```

Comment:
Thank you for using apierr :)

Reply:
You are welcome. Copy pasted this piece, but I'll take the credit :)